src/scripts/generate-test-patient-spreadsheets.sh

```diff
@@ -0,0 +1,186 @@
+    done
+  done
+}
+
+createSpreadsheetRowsForResource() {
+  local resourceType="${1}" outputCsv="${2}" curlResponse="${3}" patientId="${4}" rowNonResourceJqFields="${5}" resourceJqFields="${6}" baseUrl jqFieldValues url numRecords
+  baseUrl=$(jq -r '.baseUrl' "${TEMPLATE_FILE}")
+  declare -A jqFieldValues
+
+  url="${baseUrl}/${resourceType}?patient=${patientId}"
+  while [ -n "${url:-}" ]; do
+    jqFieldValues=()
+    request "${url}" "${curlResponse}"
+    unset url
+
+    numRecords=$(jq -r '.entry | length' "${curlResponse}")
+
+    if [ "${numRecords}" -eq 0 ]; then
+      log "INFO" "No ${resourceType} resources found for patient ${patientId}"
+      continue
+    fi
+
+    for jqField in ${resourceJqFields}; do
+      if [[ " $(jq -r ".resources[] | select( .type==\"${resourceType}\" ) | .jqFields[].name" "${TEMPLATE_FILE}" | tr '\n' ' ')" =~ [[:space:]]${jqField}[[:space:]] ]]; then
+        jqFieldValues["${jqField}"]=$(jq -r ".entry[].resource | $(jq -r ".resources[] | select(.type==\"${resourceType}\") | .jqFields[] | select(.name==\"${jqField}\") | .path" "${TEMPLATE_FILE}")" "${curlResponse}")
+      fi
+    done
+
+    for i in $(seq 1 ${numRecords}); do
+      row="${rowNonResourceJqFields}"
+      for jqField in ${resourceJqFields}; do
+        if [[ " $(jq -r " .resources[] | select( .type==\"${resourceType}\" ) | .jqFields[].name" "${TEMPLATE_FILE}" | tr '\n' ' ')" =~ [[:space:]]${jqField}[[:space:]] ]]; then
+          row+=",$(echo "${jqFieldValues[${jqField}]}" | sed -n "${i} p" | tr ',' ' ')"
+        else
+          row+=","
+        fi
+      done
+      echo "${row}" >> "${outputCsv}"
+    done
+
+    url=$(jq -r '.link[]? | select( .relation=="next" ) | .url' "${curlResponse}")
+  done
+}
+
+convertToXlsx() {
+  local inputCsv="${1}" outputXlsx="${2}"
+
+  log "INFO" "Converting csv to xlsx"
+  libreoffice --headless --convert-to xlsx --outdir "$(dirname "${outputXlsx}")" "${inputCsv}"
+}
+
+new-token() {
+  if [ "ccg" == "$(jq -r '.authentication.type' "${TEMPLATE_FILE}")" ]; then
+    new-ccg-token
+  else
+    log "ERROR" "Unsupported authentication type in template file: $(jq -r '.authentication.type' "${TEMPLATE_FILE}")"
+    exit 1
+  fi
+}
+
+new-ccg-token() {
+  local clientIdEnvVar clientSecretEnvVar audienceEnvVar oauthUrlEnvVar launchPatientEnvVar
+  clientIdEnvVar="$(jq -r '.authentication.clientIdEnvVar' "${TEMPLATE_FILE}")"
+  clientSecretEnvVar="$(jq -r '.authentication.clientSecretEnvVar' "${TEMPLATE_FILE}")"
+  audienceEnvVar="$(jq -r '.authentication.audienceEnvVar' "${TEMPLATE_FILE}")"
+  oauthUrlEnvVar="$(jq -r '.authentication.oauthUrlEnvVar' "${TEMPLATE_FILE}")"
+  launchPatientEnvVar="$(jq -r '.authentication.launchPatientEnvVar' "${TEMPLATE_FILE}")"
+  bash <(curl -sH"Authorization: Bearer $GITHUB_TOKEN" "https://raw.githubusercontent.com/department-of-veterans-affairs/shanktopus/master/bin/system-authorization-token") \
+    --client-id "${!clientIdEnvVar}" \
```

Comment:
Interesting. I knew this existed, but I've never done it in practice. Should be interesting to see.